query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -99,24 +99,27 @@ final case class PeriodicSamplesMapper(start: Long,
  * However, note that sliding window iterators have to do twice as much work, adding and removing, so it would
  * only be worth it if the overlap is more than 50%.
  */
+//scalastyle:off null
 class ChunkedWindowIterator(rv: RawDataRangeVector,
                             start: Long,
                             step: Long,
                             end: Long,
                             window: Long,
                             rangeFunction: ChunkedRangeFunction,
-                            queryConfig: QueryConfig)
-                           (windowIt: WindowedChunkIterator =
-                              new WindowedChunkIterator(rv.chunkInfos(start - window, end), start, step, end, window)
-                           ) extends Iterator[TransientRow] with StrictLogging {
+                            queryConfig: QueryConfig) extends Iterator[TransientRow] with StrictLogging {
+  private var windowIt: WindowedChunkIterator = null
```

Comment:
This is actually a case where we should just use lazy val here.  Then there is no need to use null and turn off the scalastyle.  I think it would be safer code too, and prevent others from accidentally changing the value (which all mutable variables have a risk of).

    lazy val windowIt = {
      val it = new WindowedChunkIterator(rv.chunkInfos(start - window, end), start, step, end, window)
      it.lock()
      it
    }

Reply:
Ok.